File: Data Structures & Algorithms/balanced-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private boolean balanced = true;
    public boolean isBalanced(TreeNode root) {
        if(root==null){
            return true;
        }

        dfsHeight(root);

        return balanced;
    }

    int dfsHeight(TreeNode root){

        if(root==null){
            return 0;
        }

     
            int leftHeight= dfsHeight(root.left);
            int rightHeight =dfsHeight(root.right) ;

            int gap=Math.abs(leftHeight-rightHeight);

            if(gap>1){
               this.balanced=false; 
            }

            return Math.max(leftHeight,rightHeight)+1;
        
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:

        if root == None:
            return 0
        q = [(root,1)]
        maxx = 0

        while q:
            node,lvl = q.pop(0)
            maxx = max(maxx,lvl)

            if node.left:
                q.append((node.left,lvl+1))

            if node.right:
                q.append((node.right, lvl+1))

        return maxx
        